lib/datasource/terraform-module/index.ts

```diff
@@ -127,16 +81,63 @@ export async function getReleases({
     if (latestVersion) {
       latestVersion.releaseTimestamp = res.published_at;
     }
+
     logger.trace({ dep }, 'dep');
-    const cacheMinutes = 30;
-    await packageCache.set(cacheNamespace, pkgUrl, dep, cacheMinutes);
     return dep;
-  } catch (err) {
+  }
+
+  @cache({
+    namespace: 'terraform-service-discovery',
+    key: (registryUrl: string) => `${registryUrl}/.well-known/terraform.json`,
+    ttlMinutes: 1440,
+  })
+  async getTerraformServiceDiscoveryResult(
+    registryUrl: string
+  ): Promise<ServiceDiscoveryResult> {
+    const discoveryURL = `${registryUrl}/.well-known/terraform.json`;
+    const serviceDiscovery = (
+      await this.http.getJson<ServiceDiscoveryResult>(discoveryURL)
+    ).body;
+    return serviceDiscovery;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  override handleSpecificErrors(err: HttpError): void {
     const failureCodes = ['EAI_AGAIN'];
     // istanbul ignore if
     if (failureCodes.includes(err.code)) {
       throw new ExternalHostError(err);
     }
-    throw err;
+  }
+
+  private static getRegistryRepository(
+    lookupName: string,
+    registryUrl: string
+  ): RegistryRepository {
+    let registry: string;
+    const split = lookupName.split('/');
+    if (split.length > 3 && split[0].includes('.')) {
+      [registry] = split;
+      split.shift();
+    } else {
+      registry = registryUrl;
+    }
+    if (!/^https?:\/\//.test(registry)) {
+      registry = `https://${registry}`;
+    }
+    const repository = split.join('/');
+    return {
+      registry,
+      repository,
+    };
+  }
+
+  private static getCacheKey({
```

Comment:
@viceice I also attempted to use [the new private class elements feature](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html#ecmascript-private-class-elements) here too, but I got an error when running jest. It seems like `ts-jest` needs to add support here.